lib/src/realm_class.dart

```diff
@@ -112,11 +113,22 @@ class Realm implements Finalizable {
   /// and will not update when writes are made to the database.
   late final bool isFrozen = realmCore.isFrozen(this);
 
+  /// Returns true if the [Realm] is opened for the first time and the realm file is created.
+  late final bool _openedFirstTime;
+
   /// Opens a `Realm` using a [Configuration] object.
   Realm(Configuration config) : this._(config);
 
-  Realm._(this.config, [RealmHandle? handle, this._isInMigration = false]) : _handle = handle ?? _openRealm(config) {
+  Realm._(this.config, [RealmHandle? handle, this._isInMigration = false])
+      : _openedFirstTime = !File(config.path).existsSync(),
+        _handle = handle ?? _openRealm(config) {
     _populateMetadata();
+    if (config is FlexibleSyncConfiguration) {
+      final flxConfig = config as FlexibleSyncConfiguration;
+      if (flxConfig.initialSubscriptionsConfiguration?.callback != null && (_openedFirstTime || flxConfig.initialSubscriptionsConfiguration!.rerunOnOpen)) {
```

Comment:
This `if` is duplicated below. 
How about we move this to a helper private method and named it `_shouldInvokeInitialSubscriptions` which takes the config as parameter.